daemon/api_snap_conf.go

```diff
@@ -91,6 +102,72 @@ func getSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	return SyncResponse(currentConfValues)
 }
 
+// pruneExperimentalFlags returns an exact copy of val after hiding experimental
+// features that are no longer supported from core.experimental configuration
+// generic queries (i.e. key is "" or "experimental").
+// For exact queries (e.g. core.experimental.old-flag) a client.ErrorKindConfigNoSuchOption
+// error is returned.
+//
+// This helper should only be called for core configurations. Any errors when parsing
+// core config are ignored and val is returned without modification.
+func pruneExperimentalFlags(key string, val interface{}) (interface{}, *apiError) {
+	if val == nil {
+		return val, nil
+	}
+
+	if strings.HasPrefix(key, "experimental.") {
```

Comment:
I was wondering if it would be better to only prune these if the request was not specific to any single feature flag. Just in case there are snaps out there that gate some behaviour behind a flag check (if so, it probably gets that specific config path and not all). But maybe that's overly paranoid

Reply:
That's a very good point, I like it. @pedronis would be great to know your opinion on this suggested behavior, It should be more backward compatible and safer.